### PR TITLE
Check list enable/disable persistence support

### PR DESCRIPTION
**Pull request by penenadpi**
Applying this commit:
- support for check enable/disable as part of persistent project configuration

### CODE REVIEW

File: src/iac_scan_runner/scan_runner.py [hunk 1]
```diff
@@ -222,6 +224,68 @@ def _run_checks(self, selected_checks: Optional[List], projectid: str, scan_resp
                         non_compatible_checks.append(iac_check.name)
                         write_string_to_file(iac_check.name, dir_name, "No files to scan")
                         self.results_summary.summarize_no_files(iac_check.name)
+
```
Comment (anzoman) [inline src/iac_scan_runner/scan_runner.py:227]: There is a lot of new code starting with this line. Could you extract it to a separate function?

File: src/iac_scan_runner/scan_runner.py [hunk 2]
```diff
@@ -249,37 +313,48 @@ def _run_checks(self, selected_checks: Optional[List], projectid: str, scan_resp
...
-        if check_name in self.iac_checks.keys():
-            check = self.iac_checks[check_name]
-            if not check.enabled:
-                check.enabled = True
-                return f"Check: {check_name} is now enabled and available to use."
+        if projectid and self.users_enabled:
+            self.scan_project.add_check(projectid, check_name)
+            active_project = self.scan_project.load_project(projectid)            
+            enabled_checks = active_project["checklist"]
+
```
Comment (anzoman) [inline src/iac_scan_runner/scan_runner.py:326]: This line can be removed.

**Review by anzoman — CHANGES_REQUESTED** [review]

File: src/iac_scan_runner/scan_runner.py (hunk 2 above)
Comment (penenadpi) [reply src/iac_scan_runner/scan_runner.py:326]: Fixed

File: src/iac_scan_runner/results_summary.py [hunk 3]
```diff
@@ -216,7 +216,7 @@ def dump_outcomes(self, file_name: str):
     def evaluate_verdict(self):
         verdict = "Passed"
         for check in self.outcomes:
-            if check != "uuid" and check != "time" and check != "problems" and check != "passed" and check != "total":       
+            if check != "uuid" and check != "time" and check != "problems" and check != "passed" and check != "total" and check != "execution-duration" and check != "projectid" and check!="archive":
```
[the remark this answers is not in the document]
Comment (penenadpi) [reply src/iac_scan_runner/results_summary.py:219]: Wrote a new function checking if string is not in a list

File: src/iac_scan_runner/scan_runner.py (hunk 1 above)
Comment (penenadpi) [reply src/iac_scan_runner/scan_runner.py:227]: Gonna do it for next MR

File: src/iac_scan_runner/scan_runner.py [hunk 4]
```diff
@@ -222,6 +224,68 @@ def _run_checks(self, selected_checks: Optional[List], projectid: str, scan_resp
...
+        if selected_checks and self.project_checklist:
+            for selected_check in selected_checks:
+                check = self.iac_checks[selected_check]
+                if check.name in self.project_checklist:
+                    if selected_check in compatible_checks:
+                        check_output = check.run(self.iac_dir)
+                        scan_output[selected_check] = check_output.to_dict()                        
+                        write_string_to_file(check.name, dir_name, scan_output[check.name]["output"])
+                        self.results_summary.summarize_outcome(selected_check, scan_output[check.name]["output"], self.compatibility_matrix.scanned_files, Compatibility.compatibility_matrix)
```
[the remark this answers is not in the document]
Comment (penenadpi) [reply src/iac_scan_runner/scan_runner.py]: Should be fine now, as I split it into several lines in each similar case with more than 120 chars.

**Review by anzoman — APPROVED** [review]

**Pull request merged**